playwright/e2e/create-colony.spec.ts

```diff
@@ -0,0 +1,388 @@
+      ).toBeVisible();
+    });
+
+    test.describe('Colony Name and URL validation', () => {
+      const validData = [
+        { field: 'colony name', values: validColonyNameFieldValues },
+        { field: 'custom colony URL', values: validColonyURLFieldValues },
+      ];
+
+      const invalidData = [
+        { field: 'colony name', values: invalidColonyNameFieldValues },
+        { field: 'custom colony URL', values: invalidColonyURLFieldValues },
+      ];
+
+      for (const { field, values } of validData) {
+        test(`Should accept valid ${field}`, async ({ page }) => {
+          const inputField = page.getByLabel(new RegExp(field, 'i'));
+
+          for (const value of values) {
+            await inputField.fill(value);
+            await inputField.blur();
+
+            await expect(page.getByTestId('form-error')).toBeHidden();
+          }
+        });
+      }
+
+      for (const { field, values } of invalidData) {
+        test(`Should reject invalid ${field}`, async ({ page }) => {
+          const inputField = page.getByLabel(new RegExp(field, 'i'));
+
+          for (const value of values) {
+            await inputField.fill(value);
+            await inputField.blur();
+
+            await expect(page.getByTestId('form-error')).toBeVisible();
+          }
+        });
+      }
+    });
+  });
+
+  test.describe('Native Token step', () => {
+    test.beforeEach(async ({ page }) => {
+      await fillColonyNameStep(page, {
+        nameFieldValue: colonyName,
+        urlFieldValue: customColonyURL,
+      });
+    });
+
+    test('Should render Native Token step correctly', async ({ page }) => {
+      await expect(
+        page.getByRole('heading', {
+          name: /Creating a new native token or use existing?/i,
+        }),
+      ).toBeVisible();
+
+      await expect(
+        page.getByRole('button', { name: /continue/i }),
+      ).toBeVisible();
+
+      await expect(page.getByRole('button', { name: /back/i })).toBeVisible();
+
+      await expect(page.getByLabel(/Create a new token/i)).toBeVisible();
+
+      await expect(page.getByLabel(/Use an existing token/i)).toBeVisible();
+    });
+
+    test('Should accept New Colony Token data and navigate to new step', async ({
+      page,
+    }) => {
+      const tokenName = 'Test Token Name';
+      const tokenSymbol = 'TTN';
+
+      await page.getByLabel(/Create a new token/i).check();
+      await page.getByRole('button', { name: /continue/i }).click();
+
+      await expect(
+        page.getByRole('heading', { name: /Your Colony's native token/i }),
+      ).toBeVisible();
+
+      await expect(page.getByLabel(/Create a new token/)).toBeChecked();
+
+      await expect(page.getByLabel(/Use an existing token/i)).not.toBeChecked();
+
+      await page.getByLabel(/token name/i).fill(tokenName);
+
+      await page.getByLabel(/token symbol/i).fill(tokenSymbol);
+    });
+
+    test('Token Logo file uploader should work correctly', async ({ page }) => {
```

Comment:
Please also test going back and forth and ensure the token name, symbol and logo persist

Reply:
added for the token name and symbol. for the logo I will add a test once it's fix